spec/controllers/service_controller_spec.rb

```diff
@@ -378,5 +378,30 @@
     end
   end
 
+  context 'clicking on Active/Retired Services node in the tree, after applying a filter' do
+    describe '#get_node_info' do
+      before do
+        allow(controller).to receive(:session).and_return(:adv_search => {"Service" => {:expression => {}}}, :edit => {:expression => {}})
+        controller.instance_variable_set(:@_params, :action => 'tree_select')
+        controller.instance_variable_set(:@sb, {})
+      end
+
+      it 'resets session to same values as first time in, for Active Services' do
+        controller.send(:get_node_info, 'xx-asrv')
+        reset_session_values
+      end
+
+      it 'resets session to same values as first time in, for Retired Services' do
+        controller.send(:get_node_info, 'xx-rsrv')
+        reset_session_values
+      end
+    end
+  end
+
+  def reset_session_values
```

Comment:
Please don't define methods in tests if it's not absolutely necessary. Having a duplication in this case is totally fine IMO.

Reply:
I see methods in test in many places where I consider it as less necessary than here :D ;) But, yeah, I will change this! ;)